tests/integration/baseImage_test.go

```diff
@@ -0,0 +1,97 @@
+package integration
+
+import (
+	"fmt"
+	"github.com/lf-edge/eden/pkg/controller/einfo"
+	"github.com/lf-edge/eve/api/go/config"
+	"testing"
+	"time"
+)
+
+//TestBaseImage test base image loading into eve
+func TestBaseImage(t *testing.T) {
+	ctx, err := controllerPrepare()
+	if err != nil {
+		t.Fatal("Fail in controller prepare: ", err)
+	}
+	var baseImageTests = []struct {
+		dataStoreID       string
+		imageID           string
+		baseID            string
+		imageRelativePath string
+		imageFormat       config.Format
+		eveBaseRef        string
+		zArch             string
+		HV                string
+	}{
+		{eServerDataStoreID,
+
+			"1ab8761b-5f89-4e0b-b757-4b87a9fa93ec",
+
+			"22b8761b-5f89-4e0b-b757-4b87a9fa93ec",
+
+			"../../installer/rootfs.img", // the first .. required to enforce our own FS view
+			config.Format_RAW,
+			eveBaseTag,
+			zArch,
+			eveHV,
+		},
+	}
+	for _, tt := range baseImageTests {
+		baseOSVersion := fmt.Sprintf("%s-%s", tt.eveBaseRef, tt.zArch)
+		if tt.HV != "" {
+			baseOSVersion = fmt.Sprintf("%s-%s-%s", tt.eveBaseRef, tt.zArch, tt.HV)
+		}
+		t.Run(baseOSVersion, func(t *testing.T) {
+
+			err = prepareBaseImageLocal(ctx, tt.dataStoreID, tt.imageID, tt.baseID, tt.imageRelativePath, tt.imageFormat, baseOSVersion)
+
+			if err != nil {
+				t.Fatal("Fail in prepare base image from local file: ", err)
+			}
+			deviceCtx, err := ctx.GetDeviceFirst()
+			if err != nil {
+				t.Fatal("Fail in get first device: ", err)
+			}
+			deviceCtx.SetBaseOSConfig([]string{tt.baseID})
+			devUUID := deviceCtx.GetID()
+			err = ctx.ConfigSync(deviceCtx)
+			if err != nil {
+				t.Fatal("Fail in sync config with controller: ", err)
+			}
+			t.Run("Started", func(t *testing.T) {
+				err := ctx.InfoChecker(devUUID, map[string]string{"devId": devUUID.String(), "shortVersion": baseOSVersion}, einfo.ZInfoDevSW, 300)
+				if err != nil {
+					t.Fatal("Fail in waiting for base image update init: ", err)
+				}
+			})
+			t.Run("Downloaded", func(t *testing.T) {
+				err := ctx.InfoChecker(devUUID, map[string]string{"devId": devUUID.String(), "shortVersion": baseOSVersion, "downloadProgress": "100"}, einfo.ZInfoDevSW, 1500)
+				if err != nil {
+					t.Fatal("Fail in waiting for base image download progress: ", err)
+				}
+			})
+			t.Run("Logs", func(t *testing.T) {
+				if !checkLogs {
+					t.Skip("no LOGS flag set - skipped")
+				}
+				err = ctx.LogChecker(devUUID, map[string]string{"devId": devUUID.String(), "eveVersion": baseOSVersion}, 1200)
+				if err != nil {
+					t.Fatal("Fail in waiting for base image logs: ", err)
+				}
+			})
+			timeout := time.Duration(1200)
+
+			if !checkLogs {
+				timeout = 2400
```

Comment:
Is this long time to handle qemu without ACCEL? Otherwise it shouldn't take long to locally pull 300M and reboot EVE.

BTW: are you running with timer.config.interval=5? Makes things a fair bit faster for tests.

Reply:
Seems to be long enough when I run it on my Macbook. And no -- it seems we forgot to add timer.config.interval=5 on the Eden side (as part of controller.DevModelTypeQemu) filed https://github.com/lf-edge/eden/issues/53